client/src/components/SelectAvatar.svelte

```diff
@@ -31,9 +34,11 @@
       for (const player of lobby.players) {
         newAvatarChoices[player.avatar - 1].displayName = player.displayName;
         newAvatarChoices[player.avatar - 1].available = false;
+        const playerIndex = lobby.players.indexOf(player);
+        newAvatarChoices[player.avatar - 1].uid = lobby.uids[playerIndex];
       }
 
-      if ($user !== null) {
+      if ($user !== null && lobby.uids.indexOf($user.uid) !== null) {
         const userIndex = lobby.uids.indexOf($user.uid);
```

Comment:
or just make a variable above to use for both if statement and the userIndex variable

Reply:
cant do this outside because it might be null; only using this once here so it doesn't matter